pkg/envoy/generator/cds_test.go

```diff
@@ -445,22 +482,83 @@ func TestNewResponseGetEgressTrafficPolicyNotEmpty(t *testing.T) {
 
 	ctrl := gomock.NewController(t)
-	meshCatalog := catalog.NewMockMeshCataloger(ctrl)
-	meshCatalog.EXPECT().GetInboundMeshClusterConfigs(gomock.Any()).Return(nil).Times(1)
-	meshCatalog.EXPECT().GetOutboundMeshClusterConfigs(proxyIdentity).Return(nil).Times(1)
-	meshCatalog.EXPECT().IsMetricsEnabled(proxy).Return(false, nil).AnyTimes()
-	meshCatalog.EXPECT().GetEgressClusterConfigs(proxyIdentity).Return([]*trafficpolicy.EgressClusterConfig{
-		{Name: "my-cluster"},
-		{Name: "my-cluster"}, // the test ensures this duplicate is removed
-	}, nil).Times(1)
-	meshCatalog.EXPECT().GetMeshConfig().AnyTimes()
-	meshCatalog.EXPECT().ListServicesForProxy(proxy).Return(nil, nil).AnyTimes()
-	meshCatalog.EXPECT().GetTelemetryConfig(proxy).Return(models.TelemetryConfig{}).AnyTimes()
+	mock := compute.NewMockInterface(ctrl)
+	stop := make(chan struct{})
+	meshCatalog := catalog.NewMeshCatalog(
+		mock,
+		tresorFake.NewFake(time.Hour),
+		stop,
+		messaging.NewBroker(stop),
+	)
+
+	egressPolicies := []*policyv1alpha1.Egress{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "my-cluster",
+				Namespace: "test",
+			},
+			Spec: policyv1alpha1.EgressSpec{
+				Sources: []policyv1alpha1.EgressSourceSpec{
+					{
+						Kind:      "ServiceAccount",
+						Name:      "sa-1",
+						Namespace: "test",
+					},
+					{
+						Kind:      "ServiceAccount",
+						Name:      "sa-2",
+						Namespace: "test",
+					},
+				},
+				Hosts: []string{"my-cluster"},
+				Ports: []policyv1alpha1.PortSpec{
+					{
+						Number:   80,
+						Protocol: "http",
+					},
+				},
+			},
+		},
+		// the test ensures this duplicate is removed
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "my-cluster",
+				Namespace: "test",
+			},
+			Spec: policyv1alpha1.EgressSpec{
+				Sources: []policyv1alpha1.EgressSourceSpec{
+					{
+						Kind:      "ServiceAccount",
+						Name:      "sa-1",
+						Namespace: "test",
+					},
+					{
+						Kind:      "ServiceAccount",
+						Name:      "sa-2",
+						Namespace: "test",
+					},
+				},
+				Hosts: []string{"my-cluster"},
+				Ports: []policyv1alpha1.PortSpec{
+					{
+						Number:   80,
+						Protocol: "http",
+					},
+				},
+			},
+		},
+	}
+	mock.EXPECT().GetMeshConfig().AnyTimes()
+	mock.EXPECT().ListTrafficTargets().Return(nil).AnyTimes()
+	mock.EXPECT().ListServicesForProxy(proxy).Return(nil, nil).AnyTimes()
+	mock.EXPECT().ListEgressPoliciesForServiceAccount(proxyIdentity.ToK8sServiceAccount()).Return(egressPolicies).AnyTimes()
+	mock.EXPECT().IsMetricsEnabled(proxy).Return(false, nil).AnyTimes()
+	mock.EXPECT().GetTelemetryConfig(proxy).Return(models.TelemetryConfig{}).AnyTimes()
 
 	g := NewEnvoyConfigGenerator(meshCatalog, nil)
 
 	resources, err := g.generateCDS(context.Background(), proxy)
 	tassert.NoError(t, err)
 	tassert.Len(t, resources, 1)
-	tassert.Equal(t, resources[0].(*xds_cluster.Cluster).Name, "my-cluster")
+	tassert.Equal(t, resources[0].(*xds_cluster.Cluster).Name, "my-cluster:80")
```

Comment:
tl;dr: The original output for this test was based on a mocked/hardcoded output of GetEgressClusterConfigs() that returned a config with cluster name "my-cluster" [here](https://github.com/openservicemesh/osm/blob/main/pkg/envoy/generator/cds_test.go#L453). In reality, when an egresspolicy is passed in with no protocol specified, and a config is actually generated, the cluster name will always be constructed as a host/port combination.

Since we are no longer mocking the GetEgressClusterConfigs() method, the [real](https://github.com/openservicemesh/osm/blob/main/pkg/catalog/egress.go#L28) method is called on the egressPolicies we create in what is now line 494 in the diff. In line 50 of egress.go, this calls buildClusterConfigs(), which constructs the cluster name to always reference the policy's host and port:

https://github.com/openservicemesh/osm/blob/a37d1785fef8a5488570c6adad4f7415600b6890/pkg/catalog/egress.go#L203-L208

Similar changes can be seen in rds_test.go where a lot of the mock outputs we had are no longer valid/real outputs of OSM methods